eZ/Publish/Core/REST/Server/Input/Parser/Query.php

```diff
@@ -104,4 +100,41 @@ private function processCriteriaArray(array $criteriaArray, ParsingDispatcher $p
 
         return (count($criteria) === 1) ? $criteria[0] : new CriterionValue\LogicalAnd($criteria);
     }
+
+    /**
+     * Handles SortClause data.
+     *
+     * @param array $sortClausesArray
+     * @param \eZ\Publish\Core\REST\Common\Input\ParsingDispatcher $parsingDispatcher
+     *
+     * @return array
+     */
+    private function processSortClauses(array $sortClausesArray, ParsingDispatcher $parsingDispatcher)
+    {
+        $sortClauses = [];
+
+        foreach ($sortClausesArray as $sortClauseName => $sortClauseData) {
+            if (is_array($sortClauseData) && isset($sortClauseData[0])) {
+                foreach ($sortClauseData as $singleSortClauseData) {
+                    $sortClauses[] = [
+                        'name' => $sortClauseName,
+                        'data' => $singleSortClauseData,
+                    ];
+                }
+            } else {
+                $sortClauses[] = [
+                    'name' => $sortClauseName,
+                    'data' => $sortClauseData,
+                ];
+            }
+        }
+
+        $processedSortClauses = [];
+
+        foreach ($sortClauses as $sortClause) {
+            $processedSortClauses[] = $this->dispatchSortClause($sortClause['name'], $sortClause['data'], $parsingDispatcher);
```

Comment:
any reason why this is not just done inside the first array(s) ?

Reply:
Can you elaborate, please? By the first array, you mean without additional processSortClauses method?

Reply:
Unless I am wrong, that is what @andrerom had in mind.
 ```    
 foreach ($sortClausesArray as $sortClauseName => $sortClauseData) {
    if (is_array($sortClauseData) && isset($sortClauseData[0])) {
        foreach ($sortClauseData as $singleSortClauseData) {
             $sortClauses[] = $this->dispatchSortClause($sortClauseName, $singleSortClauseData, $parsingDispatcher);
         }
      } else {
            $sortClauses[] = $this->dispatchSortClause($sortClauseName, $sortClauseData, $parsingDispatcher);
       }
    }
```

Reply:
@ViniTou that was my first aproach, but after @alongosz suggestion I refactored it, so `dispatchSortClause` is called inside foreach for normalized array (see here: https://github.com/ezsystems/ezpublish-kernel/pull/2355#discussion_r196321068 ).
My tought was this makes code better for future possible changes.

Reply:
yeah sorry, for late drive by, but @ViniTou got what I meant. The extra loop doesn't look like it does anything right now, just accumulating memory ;) :P